index.js

```diff
@@ -82,12 +83,13 @@ class K8sExecutor extends Executor {
 
         request.get(options, (err, resp, body) => {
             if (err) {
-                return response(new Error(`Error getting pod with sdbuild=${config.buildId}`));
+                return response(boom.wrap(err));
             }
             const podName = hoek.reach(body, 'items.0.metadata.name');
 
             if (!podName) {
-                return response(new Error(`Error getting pod name: ${JSON.stringify(body)}`));
+                return response(boom.notFound(`Pod with sdbuild=${config.buildId} does not exist:
```

Comment:
I would rather be less detailed here, something like `Unable to find build matching ${config.buildId}`